ansible/roles/ocp4-workload-idm/tasks/remove_workload.yml

```diff
@@ -0,0 +1,62 @@
+---
+# Implement your Workload removal tasks here
+
+# Find IP of bastion
+- name: Gather VPC facts
+  ec2_vpc_net_facts:
+    filters:
+      tag:Name: "{{ aws_vpc_name }}"
+    region: "{{ aws_region }}"
+  register: vpc
+
+- name: Get instance facts
+  ec2_instance_facts:
+    filters:
+# FIXME - find a better way to discover the bastion name
+      "tag:Name": "{{ instances[0].name }}"
+    region: "{{ aws_region }}"
+  register: instancesr
+  failed_when: instancesr|length == 0
+
+- name: Get non-terminated instance
+  set_fact:
+    ec2instance: "{{ item }}"
+  with_items:
+  - "{{ instancesr.instances }}"
+  when: not item.state.name == 'terminated'
+  failed_when: ec2instance is undefined
+
+# Remove external DNS of IdM in OpenShift cluster private zone
+- name: Get cluster metadata
+  slurp:
+    path: "{{ cluster_name }}/metadata.json"
+  register: metadata
+
+- name: Get cluster infrastructure ID
+  set_fact:
+    cluster_vpc_name: "{{ metadata.content | b64decode | from_json | json_query('infraID')}}-vpc"
+
+- name: Gather Cluster VPC facts
+  ec2_vpc_net_facts:
+    filters:
+      tag:Name: "{{ cluster_vpc_name }}"
+    region: "{{ aws_region }}"
+  register: cluster_vpc
+  failed_when: cluster_vpc.vpcs | length == 0
+
+- name: Remove Private DNS Entry for IdM in cluster private zone
+  route53:
+    state: absent
+    private_zone: yes
+    record: "{{ idm_dns_name }}"
+    type: A
+    ttl: 60
```

Comment:
can you make TTL a variable (with a default) and use it in creation and deletion ?
60 sounds a bit low. If it's not going to change, then we can maybe use 3600 ?

Reply:
I am matching the TTL that the OpenShift installer is using for consistency. Do you still want it parameterized?

Reply:
Naa, it's fine